pkg/netpol/connlist/exposure_analysis.go

```diff
@@ -0,0 +1,151 @@
+package connlist
+
+import (
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/internal/common"
+)
+
+// exposureMap from peer to its exposure data; map that stores exposure-analysis allowed connections
+// which are computed by the connlist analyzer
+type exposureMap map[Peer]*peerExposureData
+
+// helping struct - stores exposure data for a peer
+type peerExposureData struct {
+	isIngressProtected bool
+	isEgressProtected  bool
+	ingressExposure    []*xgressExposure
+	egressExposure     []*xgressExposure
+}
+
+// ----------------------------------------------------
+// xgressExposure implements XgressExposureData interface
+type xgressExposure struct {
+	exposedToEntireCluster bool
+	namespaceLabels        map[string]string
+	podLabels              map[string]string
+	potentialConn          common.AllowedConnectivity
+}
+
+func (e *xgressExposure) IsExposedToEntireCluster() bool {
+	return e.exposedToEntireCluster
+}
+
+func (e *xgressExposure) NamespaceLabels() map[string]string {
+	return e.namespaceLabels
+}
+
+func (e *xgressExposure) PodLabels() map[string]string {
+	return e.podLabels
+}
+
+func (e *xgressExposure) PotentialConnectivity() common.AllowedConnectivity {
+	return e.potentialConn
+}
+
+// ----------------------------------------------------
+// exposedPeer implements the ExposedPeer interface
+type exposedPeer struct {
+	peer               Peer
+	isIngressProtected bool
+	isEgressProtected  bool
+	ingressExp         []*xgressExposure
+	egressExp          []*xgressExposure
+}
```

Comment:
done [commit](https://github.com/np-guard/netpol-analyzer/pull/296/commits/bb07eacd050827ac0153dba0925dc85f13987e5d)

Reply:
I meant that you can use struct embedding as in my syntax above (see https://gobyexample.com/struct-embedding ).
This way you don't need a named field `pExpData` , and you can just access the fields directly (like `ep.ingressExposure` instead of `ep.pExpData.ingressExposure` )

Reply:
done [commit](https://github.com/np-guard/netpol-analyzer/pull/296/commits/d6c810112056894058b4e8221fa613c1cbb59f59) , used pointer with struct embedded (seems to me preferable)